src/main/java/com/powsybl/openloadflow/ac/outerloop/IncrementalShuntVoltageControlOuterLoop.java

```diff
@@ -47,13 +48,31 @@ public String getName() {
         return NAME;
     }
 
+    public static List<LfBus> getControlledBuses(LfNetwork network) {
+        return network.getBuses().stream()
+                .filter(bus -> bus.isVoltageControlled(VoltageControl.Type.SHUNT))
+                .filter(bus -> bus.getVoltageControl(VoltageControl.Type.SHUNT).orElseThrow().getMergeStatus() == VoltageControl.MergeStatus.MAIN)
+                .filter(bus -> !bus.getVoltageControl(VoltageControl.Type.SHUNT).orElseThrow().isHidden(true))
+                .collect(Collectors.toList());
+    }
+
+    public static List<LfShunt> getControllerElements(LfNetwork network) {
+        return network.getBuses().stream()
+                .filter(bus -> bus.isVoltageControlled(VoltageControl.Type.SHUNT))
+                .filter(bus -> bus.getVoltageControl(VoltageControl.Type.SHUNT).orElseThrow().getMergeStatus() == VoltageControl.MergeStatus.MAIN)
+                .filter(bus -> !bus.getVoltageControl(VoltageControl.Type.SHUNT).orElseThrow().isHidden(true))
+                .flatMap(bus -> bus.getShuntVoltageControl().orElseThrow().getMergedControllerElements().stream())
+                .filter(Predicate.not(LfShunt::isDisabled))
+                .collect(Collectors.toList());
+    }
```

Comment:
These two methods `getControlledBuses `and `getControllerElements` are scanning all buses, I am a bit worried about performance consequences for large networks, because they are called in the outerloop `check `method.

I might be better to store the "static/unchanging" part of this data in the outerloop context at outerloop initialization time, i.e. store the list of shunt voltage controlled buses in the outerloop context.

Then at `check` time, use this unchanging list and apply additional filters with the "dynamic" part: isHidden & isDisabled.


to be fair: not sure if an issue, it was already the case before this PR...

Reply:
Note that the MAIN status is dynamic too and cannot be computed in the beginning.

Reply:
thanks, that's still not that bad since less that half of the buses will have controls of given type